week3/homework/src/index.js

```diff
@@ -1,3 +1,140 @@
+const app = express();
+const bodyParser = require("body-parser");
+const fs = require("fs");
+app.use(bodyParser.json());
 
-// TODO: Write the homework code in this file
+app.get('/list', function (request, response) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+      return;
+    }
+    const myToDoList = JSON.parse(data);
+    if (Object.values(myToDoList).length === 0) {
+      response.end('You have not any plan in your To Do List');
+      return;
+    }
+    response.json(myToDoList);
+  });
+});
+
+app.get('/list/:id', function (request, response) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+      return;
+    }
+    const myToDoList = JSON.parse(data);
+    if (Object.values(myToDoList).length === 0) {
+      response.end('You have not any plan in your To Do List');
+      return;
+    }
+    const index = request.params.id;
+    if (isNaN(index) === true) {
+      response.end('You have to enter a number as /list/2');
+      return;
+    }
+    response.json(myToDoList[index]);
+    console.log()
+  });
+});
+
+app.post("/add/:item", function (request, response) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+      return;
+    }
+    const myToDoList = JSON.parse(data);
+    const newToDoItem = {
+      'index': myToDoList.length,
+      'description': request.params.item
+    };
+    myToDoList.push(newToDoItem);
+    fs.writeFile('./toDo.json', JSON.stringify(myToDoList), function (error) {
+      if (error) {
+        console.error(error);
+      }
+    });
+    response.json(myToDoList);
+  });
+});
+
+app.delete('/delete', function (request, response) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+      return;
+    }
+    const myToDoList = JSON.parse(data);
+    if (Object.values(myToDoList).length === 0) {
+      response.end('I can not delete an empty list, firsty try to add sth to list.');
+      return;
+    }
+    fs.writeFile('./toDo.json', '[]', function (error) {
+      if (error) { console.log(error); }
+    });
+    response.end("You have cleared all items of to-do list")
+  });
+});
+
+app.delete('/delete/:item', function (request, response) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+      return;
+    }
+    const myToDoList = JSON.parse(data);
+    if (request.params.item < 0) {
+      response.end('The parameter you entered as index is invalid');
+      return;
+    }
+    if (request.params.item > Object.values(myToDoList).length) {
+      response.end(`There is not any item in the list in the index of " ${request.params.item} "`)
+      return
+    }
+    myToDoList.splice(request.params.item, 1);
+    myToDoList.forEach(function (x) {
+      x.index = myToDoList.indexOf(x);
```

Comment:
Nice that you've thought about this! Very good! 
However, you are now resetting the index of an object in an array to the index of that array. You could remove the 'index' altogether from the objects and just use the array index. For example, if you have an array with objects in it like:
```
[
    {task: "do something"},
    {task: "do something else"},
    {task: "do something totally new"}
]
```
and want to get the first item of an array, you could use `myToDoList[0]` (you'll get `{task: "do something"}`).
If you remove this item from the array (`jsonData.splice(0,1)`), the second item becomes the first and the third becomes the second, but the id's of the array are already handeling that! 
If you now call `myToDoList[0]` you receive the new first (`{task: "do something else"}`)!

However, I think it is way more important that you thought about this, and found a solution to fix this!